pkg/cli/image/manifest/manifest.go

```diff
@@ -259,7 +259,8 @@ func (m ManifestLocation) String() string {
 }
 
 // FirstManifest returns the first manifest at the request location that matches the filter function.
-func FirstManifest(ctx context.Context, from imagereference.DockerImageReference, repo distribution.Repository, filterFn FilterFunc) (distribution.Manifest, ManifestLocation, error) {
+// If srcManifest is a DeserializedManifestList, the os/arch filter determines which manifest will be returned.
+func (o *FilterOptions) FirstManifest(ctx context.Context, from imagereference.DockerImageReference, repo distribution.Repository) (distribution.Manifest, ManifestLocation, error) {
```

Comment:
See my other comment, I don't think this is a valid change because filterFn can't just arbitrarily hide duplicates with early exit.

Reply:
I've pushed a new filter func, "IncludeSingle" that returns true only w/ exact match. ptal

Reply:
ok removed the IncludeSingle function, added a bit of logic to handle regex as regex by adding `^, $` bookends